src/app/admin/pages/learning-objects/learning-objects.component.ts

```diff
@@ -215,8 +226,21 @@ export class LearningObjectsComponent
    *
    * @param direction the direction of the sort (ASC or DESC)
    */
-  sortByDate(direction: SortType) {
-    this.query = { ...this.query, sortType: direction };
+  sortByDate() {
+    if(!this.query.sortType) {
+      this.query = {
+        sortType: SortType.Descending,
+        orderBy: OrderBy.Date
+      };
+    } else if (this.query.sortType.toString() === '-1') {
+      this.query.sortType = SortType.Ascending;
+    } else if (this.query.sortType.toString() === '1') {
```

Comment:
Why are you converting to a string?

Reply:
The way `this.query.sortType` returned was really weird, since it would return a string. However, since it's of type SortType, you can't compare it like you would a string, so I had to convert `this.query.sortType` into a string then compare if it was -1 or 1. Let me know if there's a better way to address this.